crowdgezwitscher/events/views.py

```diff
@@ -93,3 +103,45 @@ class EventAPIList(generics.ListAPIView):
 class EventAPIDetail(generics.RetrieveAPIView):
     queryset = Event.objects.filter(active=True)
     serializer_class = EventSerializer
+
+
+@api_view(['GET'])
+def get_tweets(request, pk):
+    """Get tweets for Event with primary key pk.
+
+    Searches for tweets matching the Event's registered hashtags, accounts and dates. The dates form an open interval.
+    Modify TWITTER_TWEET_COUNT to change the maximum number of returned tweet IDs.
+    """
+    event = get_object_or_404(Event, pk=pk)
+    query = event.build_twitter_search_query()
+    if not query:
+        return Response({'status': 'error', 'errors': 'Twitter not or improperly configured for this event.'},
+                        status=status.HTTP_404_NOT_FOUND)
+    since = event.coverage_start.strftime('%Y-%m-%d')
+    until = (event.coverage_end + timedelta(days=1)).strftime('%Y-%m-%d')  # to get tweets including coverage_end
+    twitter = TwitterAPI(settings.TWITTER_CONSUMER_KEY,
+                         settings.TWITTER_CONSUMER_SECRET,
+                         auth_type='oAuth2')
+    res = []
+    try:
+        tweets = twitter.request('search/tweets', {'q': query,
+                                                   'count': settings.TWITTER_TWEET_COUNT,
+                                                   'since': since,
+                                                   'until': until})
+    except TwitterConnectionError:
+        logger.warning("Could not connect to Twitter.")
+        return Response(res)
+    try:
+        for tweet in tweets:
+            try:
+                res.append(tweet['id_str'])
+            except KeyError:
+                logger.warning("Got Tweet without expected fields.")
+                continue
+    except TwitterRequestError as e:
+        if e.status_code == 429:
+            logger.warning("Twitter rate limit exhausted")
+        else:
+            logger.warning("TwitterRequestError, Status Code: %d", e.status_code)
+
+    return Response(res)
```

Comment:
Ich hätte gedacht, dass in Zeile 122 ein `TwitterConnectionError` auftreten könnte und in Zeile 127 ein `TwitterRequestError` (hier wird ja `.request` aufgerufen).